### ISSUE TITLE
Should strict-component-boundaries check the current filename 

### ISSUE THREAD

**Issue by BPScott** [Shopify/eslint-plugin-shopify#119]
Occasionally components need to import their sibling components

E.g. in `src/components/Banner/Banner.ts`: importing a Button component `import Button from './Button';`

Currently this does not flag up an error. Should this be considered a boundary violation to make people change to use `import {Button} from '..'`;

//cc @cartogram

**Comment by cartogram** [comment]
@BPScott why do you prefer `import {Button} from '..'`?

**Comment by GoodForOneFare** [comment]
Unfortunate technical constraint: preferring index imports within a component hierarchy will eventually cause very, very hard to troubleshoot dependency ordering problems.

**Comment by BPScott** [comment]
It seemed odd to me that in `src/components/Banner/Banner.ts` you could import the same module in two ways and one would pass the linting rule and one would fail:

```ts
// This passes strict-component-boundaries linting
import Button from '../Button'; 

// This imports exactly the same file, but would fail strict-component-boundaries linting
import Button from '../../components/Button';
```

As the fix for the second line failing in any file outside a components directory would be "change `import Button from 'some/path/components/Button';` to `import {Button} from 'some/path/components';`" I assumed the same would be true within the the components folder and that an import from the base would be the preferable approach.

It sounds like the first version is preferred if you're within a component folder thanks to Gord's technical constraints anyway.

Writing this up in the strict-component-boundaries docs would be good for modern-javascript-newbies like myself :)

**Comment by cartogram** [comment]
@BPScott any additional documentation is welcome and appreciated if you want to open a PR, otherwise, it sounds like you got it and the rule is working as expected.

**Comment by BPScott** [comment]
@GoodForOneFare makes it sound like there's a good reason why if you're in a "components" folder you should be using `import Button from './Button';` and not using `import {Button} from '..'`.

Currently `import {Button} from '..'` does not flag up an error if you're inside a "components". Should it throw a warning and suggest that you use `import Button from './Button';` instead?

Conversely `import Button from '../../components/Button';` does throw a warning if you're inside a "components" even though the @GoodForOneFare says that this is the correct behaviour - you should be hitting the component rather than the index in this case.

---

It sounds like strict-component-boundaries should still check the current path, but for the opposite reasons that I initially considered.

**Comment by cartogram** [comment]
Yeah it could. Also feels like it could be a completely new rule.
